api-service/service/oracle_database_agreements.go

```diff
@@ -658,3 +658,39 @@ func (as *APIService) DeleteHostFromOracleDatabaseAgreement(id primitive.ObjectI
 
 	return as.Database.UpdateOracleDatabaseAgreement(*agreement)
 }
+
+func (as *APIService) DeleteHostFromOracleDatabaseAgreements(hostname string) error {
+	if err := checkHosts(as, []string{hostname}); err != nil {
+		return err
+	}
+
+	listAgreements, err := as.Database.ListOracleDatabaseAgreements()
+	if err != nil {
+		return err
+	}
+
+	for _, la := range listAgreements {
+		for i := range la.Hosts {
+			host := la.Hosts[i]
+			if host.Hostname == hostname {
+				agreement, err := as.Database.GetOracleDatabaseAgreement(la.ID)
+				if err != nil {
+					return err
+				}
+
+				agreement.Hosts = append(
+					agreement.Hosts[0:i],
+					agreement.Hosts[i+1:len(agreement.Hosts)]...)
+
+				errAgreement := as.Database.UpdateOracleDatabaseAgreement(*agreement)
+				if errAgreement != nil {
+					return err
+				}
+
+				break
```

Comment:
If you want to use the `break`, we should change the if in `if host.Hostname == hostname` and put it inside its brackets.
Else I think here it's useless and we should remove it!
Thanks